src/navier.f90

```diff
@@ -621,9 +621,9 @@ subroutine pre_correc(ux,uy,uz,ep)
           enddo
           do k=1,xsize(3)
              do i=1,xsize(1)
-                ux(i,1,k)=byx1(i,k)+dpdxy1(i,k)
+                ux(i,1,k)=byx1(i,k) !+dpdxy1(i,k)
```

Comment:
This looks like a big modification ?

Reply:
I need to check this with Jian. I didn't spot this before.

Reply:
there should be a if statement here:
-keep the pressure gradient correction if no MHD, remove them if MHD

Reply:
Ok I have split the no correction only for MHD and add the correction back in the no MHD case.